
# Software Development Standards ![GitHub last commit](https://img.shields.io/github/last-commit/USCbiostats/software-dev)

This project’s main contents are located in the project’s
[Wiki](wiki#welcome-to-the-software-development-wiki).

# USCbiostats R packages

| Name | Description | Citations |
|:---|:---|:---|
| [**AnnoQR**](https://github.com/USCbiostats/AnnoQR) | R client wrap for AnnoQ API (<https://github.com/blueOwl/AnnoQR>) |  |
| [**aphylo**](https://github.com/USCbiostats/aphylo) | Statistical inference of genetic functions in phylogenetic trees [![CRAN status](https://www.r-pkg.org/badges/version/aphylo)](https://CRAN.R-project.org/package=aphylo) [![CRAN downloads](https://cranlogs.r-pkg.org/badges/grand-total/aphylo)](https://CRAN.R-project.org/package=aphylo) |  |
| [**bayesnetworks**](https://github.com/USCbiostats/bayesnetworks) | C++ program to fit Bayesian networks, illustrated with simulated data |  |
| [**BinaryDosage**](https://github.com/USCbiostats/BinaryDosage) | Converts VCF files to a binary format [![CRAN status](https://www.r-pkg.org/badges/version/BinaryDosage)](https://CRAN.R-project.org/package=BinaryDosage) [![CRAN downloads](https://cranlogs.r-pkg.org/badges/grand-total/BinaryDosage)](https://CRAN.R-project.org/package=BinaryDosage) |  |
| [**causnet**](https://github.com/USCbiostats/causnet) | Finds a globally optimal causal network given some data as input. |  |
| [**cit**](https://github.com/USCbiostats/cit) | Causal Inference Test [![CRAN status](https://www.r-pkg.org/badges/version/cit)](https://CRAN.R-project.org/package=cit) [![CRAN downloads](https://cranlogs.r-pkg.org/badges/grand-total/cit)](https://CRAN.R-project.org/package=cit) | [33](https://scholar.google.com/scholar?cites=2897292192206854993&hl=en) |
| [**FactorGo**](https://github.com/mancusolab/FactorGo) | Software to infer latent pleiotropic components from GWAS summary data | [8](https://scholar.google.com/citations?view_op=view_citation&hl=en&user=NfbpWAoAAAAJ&cstart=20&pagesize=80&citation_for_view=NfbpWAoAAAAJ:KxtntwgDAa4C) |
| [**fdrci**](https://github.com/USCbiostats/fdrci) | Permutation-Based FDR Point and Confidence Interval Estimation [![CRAN status](https://www.r-pkg.org/badges/version/fdrci)](https://CRAN.R-project.org/package=fdrci) [![CRAN downloads](https://cranlogs.r-pkg.org/badges/grand-total/fdrci)](https://CRAN.R-project.org/package=fdrci) | [31](https://scholar.google.com/scholar?hl=en&cites=6606625444967735035) |
| [**fmcmc**](https://github.com/USCbiostats/fmcmc) | A friendly MCMC framework [![CRAN status](https://www.r-pkg.org/badges/version/fmcmc)](https://CRAN.R-project.org/package=fmcmc) [![CRAN downloads](https://cranlogs.r-pkg.org/badges/grand-total/fmcmc)](https://CRAN.R-project.org/package=fmcmc) | [13](https://scholar.google.com/scholar?cites=18235425292318567621&hl=en) |
| [**gesso**](https://github.com/USCbiostats/gesso) | Hierarchical GxE Interactions in a Regularized Regression Model [![CRAN status](https://www.r-pkg.org/badges/version/gesso)](https://CRAN.R-project.org/package=gesso) [![CRAN downloads](https://cranlogs.r-pkg.org/badges/grand-total/gesso)](https://CRAN.R-project.org/package=gesso) | [6](https://scholar.google.com/scholar?cites=PbkWjCSETP0J&hl=en) |
| [**GxEScanR**](https://github.com/USCbiostats/GxEScanR) | R version of GxEScan [![CRAN status](https://www.r-pkg.org/badges/version/GxEScanR)](https://CRAN.R-project.org/package=GxEScanR) [![CRAN downloads](https://cranlogs.r-pkg.org/badges/grand-total/GxEScanR)](https://CRAN.R-project.org/package=GxEScanR) |  |
| [**HiLDA**](https://github.com/USCbiostats/HiLDA) | An R package for inferring the mutational exposures difference between groups [![BioC build status](https://bioconductor.org/shields/build/release/bioc/HiLDA.svg)](https://bioconductor.org/packages/release/bioc/html/HiLDA.html) [![BioC downloads](https://bioconductor.org/shields/downloads/release/HiLDA.svg)](https://bioconductor.org/packages/release/bioc/html/HiLDA.html) | [8](https://scholar.google.com/scholar?cites=2767354745053140554&hl=en) |
| [**hJAM**](https://github.com/USCbiostats/hJAM) | hJAM is a hierarchical model which unifies the framework of Mendelian Randomization and Transcriptome-wide association studies. [![CRAN status](https://www.r-pkg.org/badges/version/hJAM)](https://CRAN.R-project.org/package=hJAM) [![CRAN downloads](https://cranlogs.r-pkg.org/badges/grand-total/hJAM)](https://CRAN.R-project.org/package=hJAM) | [6](https://scholar.google.com/scholar?cites=5_2ea7Z8nBIJ&hl=en) |
| [**iMutSig**](https://github.com/USCbiostats/iMutSig) | A web application to identify the most similar mutational signature using shiny | [1](https://scholar.google.com/scholar?cites=16599818872657505515&hl=en) |
| [**jsPhyloSVG**](https://github.com/USCbiostats/jsPhyloSVG) | htmlwidgets for the jsPhyloSVG JavaScript library |  |
| [**LUCIDus**](https://github.com/USCbiostats/LUCIDus) | Latent and Unknown Cluster Analysis using Integrated Data [![CRAN status](https://www.r-pkg.org/badges/version/LUCIDus)](https://CRAN.R-project.org/package=LUCIDus) [![CRAN downloads](https://cranlogs.r-pkg.org/badges/grand-total/LUCIDus)](https://CRAN.R-project.org/package=LUCIDus) | [28](https://scholar.google.com/scholar?cites=17993227897179669521&hl=en) |
| [**MergeBinaryDosage**](https://github.com/USCbiostats/MergeBinaryDosage) | R package for merging binary dosage files |  |
| [**methcon5**](https://github.com/USCbiostats/MethCon5) | R package to identify and rank CpG DNA methylation conservation along the human genome [![CRAN status](https://www.r-pkg.org/badges/version/methcon5)](https://CRAN.R-project.org/package=methcon5) [![CRAN downloads](https://cranlogs.r-pkg.org/badges/grand-total/methcon5)](https://CRAN.R-project.org/package=methcon5) | [4](https://scholar.google.com/scholar?cites=13818868965517630908&hl=en) |
| [**partition**](https://github.com/USCbiostats/partition) | A fast and flexible framework for agglomerative partitioning in R [![CRAN status](https://www.r-pkg.org/badges/version/partition)](https://CRAN.R-project.org/package=partition) [![CRAN downloads](https://cranlogs.r-pkg.org/badges/grand-total/partition)](https://CRAN.R-project.org/package=partition) | [9](https://scholar.google.com/scholar?cites=5586649186160622454&hl=en) |
| [**pfamscanr**](https://github.com/USCbiostats/pfamscanr) | An R client for EMBL-EBI’s PfamScan API |  |
| [**polygons**](https://github.com/USCbiostats/polygons) | Flexible functions for computing polygons coordinates in R |  |
| [**rphyloxml**](https://github.com/USCbiostats/rphyloxml) | Read and write phyloXML files in R |  |
| [**selectKSigs**](https://github.com/USCbiostats/selectKSigs) | Selection of K in finding the number of mutational signatures [![BioC build status](https://bioconductor.org/shields/build/release/bioc/selectKSigs.svg)](https://bioconductor.org/packages/release/bioc/html/selectKSigs.html) [![BioC downloads](https://bioconductor.org/shields/downloads/release/selectKSigs.svg)](https://bioconductor.org/packages/release/bioc/html/selectKSigs.html) |  |
| [**slurmR**](https://github.com/USCbiostats/slurmR) | slurmR: A Lightweight Wrapper for Slurm [![CRAN status](https://www.r-pkg.org/badges/version/slurmR)](https://CRAN.R-project.org/package=slurmR) [![CRAN downloads](https://cranlogs.r-pkg.org/badges/grand-total/slurmR)](https://CRAN.R-project.org/package=slurmR) | [5](https://scholar.google.com/scholar?cites=2044081763350723149&hl=en) |
| [**sushie**](https://github.com/mancusolab/sushie) | Software to perform multi-ancestry SNP fine-mapping on molecular data | [4](https://scholar.google.com/citations?view_op=view_citation&hl=en&user=NfbpWAoAAAAJ&sortby=pubdate&citation_for_view=NfbpWAoAAAAJ:p2g8aNsByqUC) |
| [**susiepca**](https://github.com/mancusolab/susiepca) | Scalable Ultra-Sparse Bayesian PCA | [2](https://scholar.google.com/citations?view_op=view_citation&hl=en&user=NfbpWAoAAAAJ&sortby=pubdate&citation_for_view=NfbpWAoAAAAJ:_xSYboBqXhAC) |
| [**xrnet**](https://github.com/USCbiostats/xrnet) | R Package for Hierarchical Regularized Regression to Incorporate External Data [![CRAN status](https://www.r-pkg.org/badges/version/xrnet)](https://CRAN.R-project.org/package=xrnet) [![CRAN downloads](https://cranlogs.r-pkg.org/badges/grand-total/xrnet)](https://CRAN.R-project.org/package=xrnet) |  |
| [**xtune**](https://github.com/USCbiostats/xtune) | An R package for Lasso and Ridge Regression with differential penalization based on prior knowledge [![CRAN status](https://www.r-pkg.org/badges/version/xtune)](https://CRAN.R-project.org/package=xtune) [![CRAN downloads](https://cranlogs.r-pkg.org/badges/grand-total/xtune)](https://CRAN.R-project.org/package=xtune) | [25](https://scholar.google.com/scholar?cites=12002990865154112222&hl=en) |

As of 2025-02-16, the packages listed here have been cited **183** times
(source: Google Scholar).

![IMAGE Logo_horiz](https://github.com/user-attachments/assets/c6c8ce78-1759-4b29-8c5b-e9c5e444d4e9)

Many of the software in the above list are products of our "IMAGE" P01 award, "Integrative Methods of Analysis for Genetic Epidemiology", as part of our Center for Statistical Genomics. More details of this can be found [here.]( https://pphs.usc.edu/center/center-for-statistical-genomics/)





To update this list, modify the file [packages.csv](packages.csv). The
`README.md` file is updated automatically using GitHub Actions, so
there’s no need to “manually” recompile the README file after updating
the list.

# Coding Standards

Here we have the coding stadards that we ask the group to follow:

1.  [Coding Standards](wiki#coding-standards)
2.  [Software Thinking](wiki/coding-standards.md#software-thinking)
3.  [Development
    Workflow](wiki/coding-standards.md#development-workflow)
4.  [Misc](wiki/coding-standards.md#misc)

We also have some direct guidelines developed as issue templates
[here](templates).

# Bioghost Server

1.  [Introduction](wiki/Bioghost-server.md#introduction)
2.  [Setup](wiki/Bioghost-server.md#setup)
3.  [Cheat Sheet](wiki/Bioghost-server.md#cheat-sheet)

# HPC in R

- [Parallel computing in R](wiki/HPC-in-R.md#parallel-computing-in-r)  
- [parallel](wiki/HPC-in-R.md#parallel)
- [iterators+foreach](wiki/HPC-in-R.md#foreach)
- [RcppArmadillo + OpenMP](wiki/HPC-in-R.md#rcpparmadillo-and-openmp)

# Happy Scientist Seminars

The Happy Scientist Seminars are educational seminars sponsored by Core
D of IMAGE, the Biostats Program Project award. This series, the “Happy
Scientist” seminar series, is aimed at providing educational material
for members of Biostats, both students and faculty, about a variety of
tools and methods that might prove useful to them. If you have any
suggestions for subjects that you would like to learn about in future,
please send email to Kim Siegmund at (<kims@usc.edu>). Our agenda will
be driven by your specific interests as far as is possible.

A list of past seminars with material can be found
[here](/happy_scientist/).
