<!---
#########################################################################
#########################################################################
NOTE: DO NOT EDIT THE README.md BY HAND. ONLY EDIT THE README.Rmd BY HAND
#########################################################################
#########################################################################
--->
# Software Development Standards ![GitHub last commit](https://img.shields.io/github/last-commit/USCbiostats/software-dev)

This project’s main contents are located in the project’s
[Wiki](wiki#welcome-to-the-software-development-wiki).

# USCbiostats R packages

As of 2022-05-23, the packages listed here have been cited **72** times
(source: Google Scholar).

| Name                                                                      | Description                                                                                                                                                                                                                                                                                                              | Citations                                                                 |
|:--------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:--------------------------------------------------------------------------|
| [**AnnoQR**](https://github.com/USCbiostats/AnnoQR)                       | R client wrap for AnnoQ API (<https://github.com/blueOwl/AnnoQR>)                                                                                                                                                                                                                                                        |                                                                           |
| [**aphylo**](https://github.com/USCbiostats/aphylo)                       | Statistical inference of genetic functions in phylogenetic trees                                                                                                                                                                                                                                                         |                                                                           |
| [**bayesnetworks**](https://github.com/USCbiostats/bayesnetworks)         | C++ program to fit Bayesian networks, illustrated with simulated data                                                                                                                                                                                                                                                    |                                                                           |
| [**BinaryDosage**](https://github.com/USCbiostats/BinaryDosage)           | Converts VCF files to a binary format                                                                                                                                                                                                                                                                                    |                                                                           |
| [**causnet**](https://github.com/USCbiostats/causnet)                     | What the Package Does (One Line, Title Case)                                                                                                                                                                                                                                                                             |                                                                           |
| [**cit**](https://github.com/USCbiostats/cit)                             | Causal Inference Test                                                                                                                                                                                                                                                                                                    | [15](https://scholar.google.com/scholar?cites=2897292192206854993&hl=en)  |
| [**fdrci**](https://github.com/USCbiostats/fdrci)                         | Permutation-Based FDR Point and Confidence Interval Estimation                                                                                                                                                                                                                                                           | [23](https://scholar.google.com/scholar?hl=en&cites=6606625444967735035)  |
| [**fmcmc**](https://github.com/USCbiostats/fmcmc)                         | A friendly MCMC framework                                                                                                                                                                                                                                                                                                | [4](https://scholar.google.com/scholar?cites=18235425292318567621&hl=en)  |
| [**gesso**](https://github.com/USCbiostats/gesso)                         | Hierarchical GxE Interactions in a Regularized Regression Model                                                                                                                                                                                                                                                          |                                                                           |
| [**GxEScanR**](https://github.com/USCbiostats/GxEScanR)                   | R version of GxEScan                                                                                                                                                                                                                                                                                                     |                                                                           |
| [**HiLDA**](https://github.com/USCbiostats/HiLDA)                         | An R package for inferring the mutational exposures difference between groups[![](https://img.shields.io/badge/Bioconductor%20version-1.0.0-green.svg)](https://www.bioconductor.org/packages/HiLDA)[![](https://img.shields.io/badge/download-3275/total-blue.svg)](https://bioconductor.org/packages/stats/bioc/HiLDA) | [3](https://scholar.google.com/scholar?cites=2767354745053140554&hl=en)   |
| [**hJAM**](https://github.com/USCbiostats/hJAM)                           | hJAM is a hierarchical model which unifies the framework of Mendelian Randomization and Transcriptome-wide association studies.                                                                                                                                                                                          | [1](https://scholar.google.com/scholar?cites=5_2ea7Z8nBIJ&hl=en)          |
| [**iMutSig**](https://github.com/USCbiostats/iMutSig)                     | A web application to identify the most similar mutational signature using shiny                                                                                                                                                                                                                                          | [1](https://scholar.google.com/scholar?cites=16599818872657505515&hl=en)  |
| [**jsPhyloSVG**](https://github.com/USCbiostats/jsPhyloSVG)               | htmlwidgets for the jsPhyloSVG JavaScript library                                                                                                                                                                                                                                                                        |                                                                           |
| [**LUCIDus**](https://github.com/USCbiostats/LUCIDus)                     | Latent and Unknown Cluster Analysis using Integrated Data                                                                                                                                                                                                                                                                | [10](https://scholar.google.com/scholar?cites=17993227897179669521&hl=en) |
| [**MergeBinaryDosage**](https://github.com/USCbiostats/MergeBinaryDosage) | R package for merging binary dosage files                                                                                                                                                                                                                                                                                |                                                                           |
| [**methcon5**](https://github.com/USCbiostats/MethCon5)                   | R package to identify and rank CpG DNA methylation conservation along the human genome                                                                                                                                                                                                                                   | [3](https://scholar.google.com/scholar?cites=13818868965517630908&hl=en)  |
| [**partition**](https://github.com/USCbiostats/partition)                 | A fast and flexible framework for agglomerative partitioning in R                                                                                                                                                                                                                                                        | [2](https://scholar.google.com/scholar?cites=5586649186160622454&hl=en)   |
| [**pfamscanr**](https://github.com/USCbiostats/pfamscanr)                 | An R client for EMBL-EBI’s PfamScan API                                                                                                                                                                                                                                                                                  |                                                                           |
| [**polygons**](https://github.com/USCbiostats/polygons)                   | Flexible functions for computing polygons coordinates in R                                                                                                                                                                                                                                                               |                                                                           |
| [**rphyloxml**](https://github.com/USCbiostats/rphyloxml)                 | Read and write phyloXML files in R                                                                                                                                                                                                                                                                                       |                                                                           |
| [**selectKSigs**](https://github.com/USCbiostats/selectKSigs)             | Selection of K in finding the number of mutational signatures[![](https://img.shields.io/badge/Bioconductor%20version-1.0.0-green.svg)](https://www.bioconductor.org/packages/selectKSigs)[![](https://img.shields.io/badge/download-2352/total-blue.svg)](https://bioconductor.org/packages/stats/bioc/selectKSigs)     |                                                                           |
| [**slurmR**](https://github.com/USCbiostats/slurmR)                       | slurmR: A Lightweight Wrapper for Slurm                                                                                                                                                                                                                                                                                  | [2](https://scholar.google.com/scholar?cites=2044081763350723149&hl=en)   |
| [**xrnet**](https://github.com/USCbiostats/xrnet)                         | R Package for Hierarchical Regularized Regression to Incorporate External Data                                                                                                                                                                                                                                           | [2](https://scholar.google.com/scholar?cites=12362903479655163477&hl=en)  |
| [**xtune**](https://github.com/USCbiostats/xtune)                         | An R package for Lasso and Ridge Regression with differential penalization based on prior knowledge                                                                                                                                                                                                                      | [6](https://scholar.google.com/scholar?cites=12002990865154112222&hl=en)  |

To update this list, modify the file [packages.csv](packages.csv). The
`README.md` file is updated automatically using GitHub Actions, so
there’s no need to “manually” recompile the README file after updating
the list.

# Coding Standards

1.  [Coding Standards](wiki#coding-standards)
2.  [Software Thinking](wiki/coding-standards.md#software-thinking)
3.  [Development
    Workflow](wiki/coding-standards.md#development-workflow)
4.  [Misc](wiki/coding-standards.md#misc)

We do have some direct guidelines developed as issue templates
[here](templates).

# Bioghost Server

1.  [Introduction](wiki/Bioghost-server.md#introduction)
2.  [Setup](wiki/Bioghost-server.md#setup)
3.  [Cheat Sheet](wiki/Bioghost-server.md#cheat-sheet)

# HPC in R

-   [Parallel computing in
    R](wiki/HPC-in-R.md#parallel-computing-in-r)  
-   [parallel](wiki/HPC-in-R.md#parallel)
-   [iterators+foreach](wiki/HPC-in-R.md#foreach)
-   [RcppArmadillo + OpenMP](wiki/HPC-in-R.md#rcpparmadillo-and-openmp)

# Happy Scientist Seminars

The Happy Scientist Seminars are educational seminars sponsored by Cores
C and D of IMAGE, the Biostats Program Project award. This series, the
“Happy Scientist” seminar series, is aimed at providing educational
material for members of Biostats, both students and faculty, about a
variety of tools and methods that might prove useful to them. If you
have any suggestions for subjects that you would like to learn about in
future, please send email to Paul Marjoram at (<pmarjora@usc.edu>). Our
agenda will be driven by your specific interests as far as is possible.

List of past seminars with material can be found
[here](/happy_scientist/).
